sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -0,0 +1,407 @@
+        /// <summary>
+        /// Add an existing client for sending messages to an event hub.  Infer the eventHub name from client.path
+        /// </summary>
+        /// <param name="client"></param>
+        public void AddEventHubProducerClient(EventHubProducerClient client)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+            string eventHubName = client.EventHubName;
+            AddEventHubProducerClient(eventHubName, client);
+        }
+
+        /// <summary>
+        /// Add an existing client for sending messages to an event hub.  Infer the eventHub name from client.path
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub</param>
+        /// <param name="client"></param>
+        public void AddEventHubProducerClient(string eventHubName, EventHubProducerClient client)
+        {
+            if (eventHubName == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            _clients[eventHubName] = client;
+        }
+
+        /// <summary>
+        /// Add a connection for sending messages to an event hub. Connect via the connection string.
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub. </param>
+        /// <param name="sendConnectionString">connection string for sending messages. If this includes an EntityPath, it takes precedence over the eventHubName parameter.</param>
+        public void AddSender(string eventHubName, string sendConnectionString)
+        {
+            if (eventHubName == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+            if (sendConnectionString == null)
+            {
+                throw new ArgumentNullException(nameof(sendConnectionString));
+            }
+
+            EventHubsConnectionStringBuilder sb = new EventHubsConnectionStringBuilder(sendConnectionString);
+            if (string.IsNullOrWhiteSpace(sb.EntityPath))
+            {
+                sb.EntityPath = eventHubName;
+            }
+
+            var client = new EventHubProducerClient(sb.ToString());
+            AddEventHubProducerClient(eventHubName, client);
+        }
+
+        /// <summary>
+        /// Add a connection for listening on events from an event hub. Connect via the connection string and use the SDK's built-in storage account.
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub</param>
+        /// <param name="receiverConnectionString">connection string for receiving messages. This can encapsulate other service bus properties like the namespace and endpoints.</param>
+        public void AddReceiver(string eventHubName, string receiverConnectionString)
```

Comment:
We probably want to move to `Consumer` to align with the new terminology.    This was a big ask from the service team during T2 API reviews.